gtda/utils/tests/test_validation.py

```diff
@@ -54,3 +56,206 @@ def test_inputs_arrayStruc_V():
 
     with pytest.raises(ValueError):
         check_diagrams(X)
+
+
+# Testing check_point_clouds
+# Create several kinds of inputs
+class CreateInputs:
+    def __init__(
+            self, n_samples, n_1, n_2, n_samples_extra, n_1_extra, n_2_extra
+    ):
+        N = n_samples * n_1 * n_2
+        n_1_rectang = n_1 + 1
+        n_2_rectang = n_2 - 1
+        N_rectang = n_samples * n_1_rectang * n_2_rectang
+
+        self.X = np.arange(N, dtype=float).reshape(n_samples, n_1, n_2)
+        self.X_rectang = np.arange(N_rectang, dtype=float). \
+            reshape(n_samples, n_1_rectang, n_2_rectang)
+
+        self.X_list = []
+        self.X_list_rectang = []
+        for i in range(n_samples):
+            self.X_list.append(self.X[i].copy())
+            self.X_list_rectang.append(self.X_rectang[i].copy())
+
+        # List example where not all 2D arrays have the same no. of rows
+        self.X_list_rectang_diff_rows = \
+            self.X_list_rectang[:-1] + [self.X_list_rectang[-1][:-1, :]]
+
+        # List example where not all 2D arrays have the same no. of columns
+        self.X_list_rectang_diff_cols = \
+            self.X_list_rectang[:-1] + [self.X_list_rectang[-1][:, :-1]]
+
+        N_extra = n_samples_extra * n_1_extra * n_2_extra
+        X_extra = np.arange(N_extra, dtype=float). \
+            reshape(n_samples_extra, n_1_extra, n_2_extra)
+        X_list_extra = []
+        for i in range(n_samples_extra):
+            X_list_extra.append(X_extra[i].copy())
+        self.X_list_tot = self.X_list + X_list_extra
```

Comment:
I don't fully understand that. Why do we need those?

Reply:
I wanted to cover the case of a list of square arrays of varying dimension.

Reply:
Yes, my bad.